api/src/org/labkey/api/data/DataRegionSelection.java

```diff
@@ -319,6 +312,36 @@ public static int setSelectionForAll(QueryForm form, boolean checked) throws IOE
         return setSelectionForAll(view, form.getQuerySettings().getSelectionKey(), checked);
     }
 
+    private static List<String> setDataRegionColumnsForSelection(DataRegion rgn, RenderContext rc, QueryView view, TableInfo table)
+    {
+        // force the pk column(s) into the default list of columns
+        List<String> selectorColNames = rgn.getRecordSelectorValueColumns();
+        if (selectorColNames == null)
+            selectorColNames = table.getPkColumnNames();
+        List<ColumnInfo> selectorColumns = new ArrayList<>();
+        for (String colName : selectorColNames)
+        {
+            if (null == rgn.getDisplayColumn(colName)) {
+                selectorColumns.add(table.getColumn(colName));
+            }
+        }
+        ActionURL url = view.getSettings().getSortFilterURL();
+
+        Sort sort = rc.buildSort(table, url, rgn.getName());
+        SimpleFilter filter = rc.buildFilter(table, rc.getColumnInfos(rgn.getDisplayColumns()), url, rgn.getName(), -1, 0, sort);

```

Comment:
It doesn't look like maxRows is used except in the NestedRenderContext, but might as well use the ALL_ROWS constant.

```suggestion
        SimpleFilter filter = rc.buildFilter(table, rc.getColumnInfos(rgn.getDisplayColumns()), url, rgn.getName(), Table.ALL_ROWS, 0, sort);
```